docker-app/qfieldcloud/core/models.py

```diff
@@ -1546,6 +1546,24 @@ def fallback_output(self) -> str:
                 "The job ended in unknown state. Please verify the project is configured properly, try again and contact QFieldCloud support for more information."
             )
 
+    def clean(self) -> None:
+        useraccount = self.created_by.useraccount
+        current_subscription = useraccount.active_subscription
+        exception_msg_prefix = "Cannot create job because "
+
+        if not current_subscription.is_active:
+            raise ValidationError(f"{exception_msg_prefix} UserAccount inactive.")
+
+        if useraccount.storage_free_bytes < 0:
+            raise ValidationError(
+                f"{exception_msg_prefix} No free storage for UserAccount."
+            )
+
+        # TODO
+        # * self.project.has_online_layers? and self.project.owner.plan.not_support_online_layers?
+
```

Comment:
@suricactus could you give me some hint how the find out whether `project has online layers (postgis)`?

Reply:
Use `project.has_online_vector_data` dynamic property.